### DISCUSSION TITLE
Using Traits in either `EpiAware` or `EpiAwarePipeline`

### DISCUSSION THREAD

**Discussion by SamuelBrand1** [CDCgov/Rt-without-renewal#297] [Ideas]
Following on from #295 I feel that the way we have done the pipeline at least might have been easier to do in a Trait based way. 

The OG of trait based programming in Julia is https://github.com/JuliaLang/julia/issues/2345#issuecomment-54537633 but its quite a popular approach, e.g. [`SimpleTraits.jl`](https://github.com/mauro3/SimpleTraits.jl) .

I'm not sure this even qualifies as an issue, but might be an interesting discussion point.

**Comment by seabbs** [comment]
Yes I have been wondering about this. It is essentially for dispatching in another dimension (so things can have a type and a trait right?)

**Reply by SamuelBrand1** [reply]
Yes, and it gives finer grain control on how types dispatch.
